pkgs/by-name/at/atlauncher/package.nix

```diff
@@ -20,27 +19,44 @@
   xorg,
 }:
 
-stdenvNoCC.mkDerivation (finalAttrs: {
```

Comment:
why get rid of finalAttrs? it would make version pinning much easier

Reply:
Can you educate me on that? The `let in` syntax is a personal preference of mine and I didn't see any notion against it yet.

Reply:
> Both `version` and `src` have to be overridden regardless of `finalAttrs` use

Yeah, that's why I don't know why is `finalAttrs` preferred.

Moreover, I am also leaning away from `finalAttrs` after reading this comment https://github.com/NixOS/nixpkgs/issues/320528#issuecomment-2175087036. I am aware that there is no conclusive evidence of this yet but it simply adds on to my existing bias.

Reply:
finalAttrs is calculated after all the overrides. Currently, version/src are used in installPhase and postInstall. With let/rec, you have to override version, src, installPhase, postInstall, and maybe meta, with finalAttrs you only have to override version/src

Reply:
> Currently, version/src are used in installPhase and postInstall. 

`version` attribute is declared as a Bash variable, perhaps it’d make more sense to use that instead of string interpolation then? E.g. `build/libs/ATLauncher-$version.jar` (notice `$version` instead of `${version}`, the latter is a Nix expression that can be escaped with `''${…}`). Same effect, but no need for `finalAttrs`.

Reply:
I'll use `finalAttrs` in that case. The build environment stuff makes sense too but I want to stick to the well known methods for now.

> With let/rec, you have to override version, src, installPhase, postInstall, and maybe meta, with finalAttrs you only have to override version/src

let-in should be added to the [hall of shame](https://nix.dev/guides/best-practices.html) in that case.